### ISSUE TITLE
Data passed to wrong activation parameter

### ISSUE THREAD

**Issue by foster999** [lithops-cloud/lithops#1201]
I've created a reproducible example for the error that I noticed in #1192:

```import joblib
import pandas as pd
from lithops.util.joblib import register_lithops
from sklearn.datasets import load_digits
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import GridSearchCV
from sklearn.pipeline import FunctionTransformer, Pipeline

data = pd.DataFrame(
    {"id": [1, 2, 3, 4, 5, 6], "features": [[1, 2, 3, 4, 5] for _ in range(6)]}
)

target = pd.Series([0, 1, 0, 1, 0, 1])
param_grid = {
    "classifier__n_estimators": [100, 50, 25],
}

# Expands nested columns
format_transformer = FunctionTransformer(lambda df: df.features.apply(pd.Series))

pipeline = Pipeline(
    [
        ("format", format_transformer),
        ("classifier", RandomForestClassifier()),
    ]
)
search = GridSearchCV(pipeline, param_grid, cv=2, refit=True)


register_lithops()
with joblib.parallel_backend("lithops"):
    search.fit(
        data,
        target,
    )

print("Best CV score: %0.3f" % search.best_score_)
print("Best parameters set:")
print(search.best_params_)
```

Running this with the config:
```
lithops:
    backend: localhost
    storage: localhost
    log_level: DEBUG
```

Gives the error:
```
Traceback (most recent call last):
  File "/private/var/folders/t4/hdpq5bd51tjfws2mq4kj7gqh0000gn/T/lithops-user/lithops/worker/jobrunner.py", line 236, in run
    result = func(**data)
    ^^^^^^^^^^^^^^^^^
  File "/private/var/folders/t4/hdpq5bd51tjfws2mq4kj7gqh0000gn/T/lithops-user/lithops/multiprocessing/process.py", line 117, in cloud_process_wrapper
    raise exception
  File "/private/var/folders/t4/hdpq5bd51tjfws2mq4kj7gqh0000gn/T/lithops-user/lithops/multiprocessing/process.py", line 92, in cloud_process_wrapper
    res = func(*data['args'], **data['kwargs'])
    ^^^^^^^^^^^^^^^^^
  File "/private/var/folders/t4/hdpq5bd51tjfws2mq4kj7gqh0000gn/T/lithops-user/lithops/util/joblib/lithops_backend.py", line 183, in handle_call_threads
    results = pool.starmap(handle_call_process, mem_opt_calls)
    ^^^^^^^^^^^^^^^^^
  File "/Users/user/.pyenv/versions/3.11.4/lib/python3.11/multiprocessing/pool.py", line 375, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
    ^^^^^^^^^^^^^^^^^
  File "/Users/user/.pyenv/versions/3.11.4/lib/python3.11/multiprocessing/pool.py", line 774, in get
    raise self._value
  File "/Users/user/.pyenv/versions/3.11.4/lib/python3.11/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
    ^^^^^^^^^^^^^^^^^
  File "/Users/user/.pyenv/versions/3.11.4/lib/python3.11/multiprocessing/pool.py", line 51, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
      ^^^^^^^^^^^^^^^^^
  File "/private/var/folders/t4/hdpq5bd51tjfws2mq4kj7gqh0000gn/T/lithops-user/lithops/util/joblib/lithops_backend.py", line 192, in handle_call_process
    return func(*args, **kwargs)
      ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/sklearn/utils/parallel.py", line 127, in __call__
    return self.function(*args, **kwargs)
    ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/sklearn/ensemble/_forest.py", line 188, in _parallel_build_trees
    tree.fit(X, y, sample_weight=curr_sample_weight, check_input=False)
    ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/sklearn/base.py", line 1152, in wrapper
    return fit_method(estimator, *args, **kwargs)
    ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/sklearn/tree/_classes.py", line 959, in fit
    super()._fit(
  File "/venv/lib/python3.11/site-packages/sklearn/tree/_classes.py", line 270, in _fit
    n_samples, self.n_features_in_ = X.shape
```

This is because the data from the `y` parameter of the last activation is also being passed to the `X` parameter, instead of the expected data.

This error doesn't occur with the `loky` backend for joblib.

I traced the data through the sklearn code and can see that [the correct data are passed to the parallel backend here](https://github.com/scikit-learn/scikit-learn/blob/f0fee2736426553a8a32df1ff153ee9ccdfb434b/sklearn/ensemble/_forest.py#L496), but are then incorrect in the parallel function call (`_parallel_build_trees`) on the lithops worker process.

**Comment by JosepSampe** [comment]
I've been exploring this and the data lithops receives is:
```
function: <sklearn.utils.parallel._FuncWrapper object at 0x7f91f2805000>
args: [DecisionTreeClassifier(max_features='sqrt', random_state=2131263703), True, array([3, 4, 5]), array([0, 1, 2]), None, 8, 100]
kwargs: {'verbose': 0, 'class_weight': None, 'n_samples_bootstrap': 6}
```

@foster999 Do you know which parameter is missing or which parameter shouldn't be here?

**Comment by foster999** [comment]
Thanks for taking a look @JosepSampe! Where in the lithops code are you viewing this, for debugging?

**Note that the issue only occurs in the last lithops activation (the one prefering threads), which is created [after this line is hit](https://github.com/scikit-learn/scikit-learn/blob/3ab6c8cf0cf6b8253d9e5a9d52908129fc147987/sklearn/model_selection/_search.py#L1008C33-L1008C33).**

X (args[2]) and y (args[3]) are the data of interest here.
The data sklearn pass to the parallel backend as args are:
```
>>> X
array([[1., 2., 3., 4., 5.],
       [1., 2., 3., 4., 5.],
       [1., 2., 3., 4., 5.],
       [1., 2., 3., 4., 5.],
       [1., 2., 3., 4., 5.],
       [1., 2., 3., 4., 5.]], dtype=float32)

>>> y
array([[0.],
       [1.],
       [0.],
       [1.],
       [0.],
       [1.]])
```

But when `_FuncWrapper` is called on the lithops worker it gets called with:
```
>>> args[2]
array([3, 4, 5])

>>> args[3]
array([0, 1, 2])
```

**Comment by JosepSampe** [comment]
@foster999 I found the source of the issue. I added the fix in #1202

**Comment by foster999** [comment]
Amazing, that's working for me. Thanks for the quick responses to all of these issues!
